### PR TITLE
CAD instead of OCC

### PR DESCRIPTION
**Pull request by jfussbro**
**_Describe your changes here:_**

With this PR we switch our extern geometry naming from occ to cad, in order to keep it more general.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: .github/workflows/tests_sc_p4est.yml [hunk 1]
```diff
@@ -103,7 +103,7 @@ jobs:
       run: echo No cache found or cache will be ignored. SC_IGNORE_CACHE=$SC_IGNORE_CACHE
     - name: if ignore cache, delete folders
       if: ${{ env.SC_IGNORE_CACHE == 1 }}
-      # The true at the end is to ignore errors that i.e. occur when the folders do not exist
+      # The true at the end is to ignore errors that i.e. cadur when the folders do not exist
```
Comment (jmark) [inline .github/workflows/tests_sc_p4est.yml:106]: @jfussbro This is not good ... please check for this before making a PR. Most text substitution tools have a parameter/switch where you tell it to only replace stand-alone words. What tool are you using?

```suggestion
      # The true at the end is to ignore errors that i.e. occur when the folders do not exist
```

**Review by jmark — CHANGES_REQUESTED** [review]

File: .github/workflows/tests_sc_p4est.yml (hunk 1 above)
Comment (jfussbro) [reply .github/workflows/tests_sc_p4est.yml]: I'm sorry. I didn't think about that. I used find and replace in VS code.

File: .github/workflows/tests_t8code_linkage_parallel_debug.yml [hunk 2]
```diff
@@ -205,25 +205,25 @@ jobs:
     - name: check OpenCASCADE
       run: echo "Checking OpenCASCADE with MPI in debugging mode"
     - name: configure MPI OpenCASCADE debug
-      run: mkdir build_occ && cd build_occ && ../configure $CONFIG_DEBUG --with-occ CFLAGS="$CFLAGS_var -I/usr/include/opencascade" CXXFLAGS="$CXXFLAGS_var -I/usr/include/opencascade"
+      run: mkdir build_cad && cd build_cad && ../configure $CONFIG_DEBUG --with-cad CFLAGS="$CFLAGS_var -I/usr/include/opencascade" CXXFLAGS="$CXXFLAGS_var -I/usr/include/opencascade"
```
Comment (jmark) [inline .github/workflows/tests_t8code_linkage_parallel_debug.yml:208]: How can this even work when the `configure` script does not accept `--with-cad`?

https://github.com/DLR-AMR/t8code/blob/44d96ab2dd2224c9a4cd935c47ef4eb108cbe0a9/config/t8_occ.m4#L15

File: .github/workflows/tests_t8code_linkage_parallel_debug.yml
Comment (jmark) [inline .github/workflows/tests_t8code_linkage_parallel_debug.yml:1]: Is it necessary to rename to `cad`? We are linking against `occ` here, right?

**Review by jmark — CHANGES_REQUESTED** [review]

**Review by jmark — APPROVED** [review]

**Pull request merged**